### DISCUSSION TITLE
Add Types for GeoGebra API

### DISCUSSION THREAD

**Discussion by hyrious** [DefinitelyTyped/DefinitelyTyped#66615] [Request a new @types package]
Me and @gwhitney wants to use [GeoGebra's API](https://wiki.geogebra.org/en/Reference:GeoGebra_Apps_API) in TypeScript, however the only information we can get from their document / [repository](https://github.com/geogebra/geogebra) is just hand-written descriptions and Java/JavaScript codes, no TypeScript definitions. So I spent some time to [generate](https://github.com/netless-io/netless-app/tree/master/packages/app-geogebra/gen) these [typings](https://github.com/netless-io/netless-app/tree/master/packages/app-geogebra/src/types) from all useful information I could find.

As I was mentioned in https://github.com/netless-io/netless-app/pull/25, there're at least 2 issues to be discussed before publishing these codes as an `@types/geogebra` package:

- There's no npm package for deployggb.js, the product is distributed through their [CDN](https://cdn.geogebra.org/apps/deployggb.js). So the new package is not linked to a real npm package, which is unusual.
- The API could make incompatible changes even in patching versions, i.e. there's no semantic versioning to this product. I guess this is mostly because they write java codes which is compiled to js, and not notice the incompatible changes.

**Plan A**:

@geogebra make some tooling to generate TypeScript definitions from their java code, which should always give us the latest precise APIs. It even does not need to be published as a package.

**Plan B**:

Add a new package `@types/geogebra` or `@types/deployggb.js` which exposes interfaces to use the GeoGebra's API. It can be maintained in this repo and the community can keep updating this package.

**Comment by gwhitney** [comment]
In the case of Plan B, I would think the better name to use would be `@types/deployggb.js` -- then in case there is ever an npm geogebra package, the name won't be in conflict. `deployggb.js` seems like a much less likely name for a npm package.
